extensions/ql-vscode/test/vscode-tests/cli-integration/variant-analysis/variant-analysis-manager.test.ts

```diff
@@ -388,4 +388,37 @@ describe("Variant Analysis Manager", () => {
       }
     }
   });
+
+  describe("runVariantAnalysisFromPublishedPack", () => {
+    it("should download pack for correct language and identify problem queries", async () => {
+      const showQuickPickSpy = jest
+        .spyOn(window, "showQuickPick")
+        .mockResolvedValue(
+          mockedQuickPickItem({
+            label: "JavaScript",
+            description: "javascript",
+            language: "javascript",
+          }),
+        );
+
+      const runVariantAnalysisMock = jest.fn();
+      variantAnalysisManager.runVariantAnalysis = runVariantAnalysisMock;
```

Comment:
I find it a bit weird that we're testing the manager but mock out parts of it. But it also makes sense, at least at this point. Perhaps in the future we can use the mock API server and have a test that covers the flow more.

Reply:
Yes, I'm also not totally sure about it. What makes it ok for me is that it will fail if the relationship between these two methods changes. We check that the mocked method was called, and that it was called with the correct arguments.

The direction it pushes me in is that maybe `runVariantAnalysisFromPublishedPack` shouldn't be a part of `VariantAnalysisManager`. It could be a method somewhere else that just consumes `VariantAnalysisManager` as a dependency. Or that `runVariantAnalysis` should be moved somewhere else that then `VariantAnalysisManager` can consume as a dependency. So we separate the definition of the entrypoint of the commands from where all the shared implementation happens.